src/components/form/favorite-modal.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Modal } from 'react-bootstrap';
+import { ModalInput } from 'hadron-react-components';
+import { TextButton } from 'hadron-react-buttons';
+import Actions from 'actions';
+
+import styles from '../connect.less';
+
+/**
+ * The favorite modal.
+ */
+class FavoriteModal extends PureComponent {
+  static displayName = 'FavoriteModal';
+
+  static propTypes = {
+    currentConnection: PropTypes.object,
+    isModalVisible: PropTypes.bool.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { name: props.currentConnection.name };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.currentConnection.name !== this.state.name) {
+      this.setState({ name: nextProps.currentConnection.name });
+    }
+  }
+
+  /**
+   * Deletes a favorite.
+   */
+  onDeleteFavoriteClicked() {
+    Actions.onDeleteConnectionClicked(this.props.currentConnection);
+    Actions.hideFavoriteModal();
+  }
+
+  /**
+   * Closes modal.
+   */
+  handleClose() {
+    Actions.hideFavoriteModal();
+  }
+
+  /**
+   * Saves the favorite.
+   */
+  handleSave() {
+    Actions.onCreateFavoriteClicked(this.state.name);
+    Actions.hideFavoriteModal();
+    setTimeout(() => Actions.showFavoriteMessage(), 800);
+    setTimeout(() => Actions.hideFavoriteMessage(), 2800);
+  }
+
+  /**
+   * Submits the form.
+   *
+   * @param {Object} evt - evt.
+   */
+  handleFormSubmit(evt) {
+    evt.preventDefault();
+    evt.stopPropagation();
```

Comment:
to double check: you want to do both of these preventions, yea?

Reply:
yes! :)